src/components/Weather.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCitiesWeatherData } from '../redux/slice/weatherSlice';
+import Nav from './Nav';
+import CityWeather from './City/CityWeather';
+import './styles/Weather.css';
+import AfricaWeather from './AfricaWeather';
+
+const Weather = () => {
+  const [filteredCities, setFilteredCities] = useState([]);
+  const cityData = useSelector((state) => state.weather.dataOfCities);
+  const isDataFetched = useSelector((state) => state.weather.isDataFetched);
+  const dispatch = useDispatch();
+
+  const handleFilterChange = (filterText) => {
+    // eslint-disable-next-line max-len
```

Comment:
- [ Required ] I think it would be better if you fix this linter error from your project, it's considered bad practice to disable linter checks in source files. Linter checks help you as a developer to write clean code and because of this it's not professional when you disable them. I would like to suggest that you refactor this line of code as shown below.

## Example code
> Please check lines 17 through to 20
```JavaScript
import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { fetchCitiesWeatherData } from '../redux/slice/weatherSlice';
import Nav from './Nav';
import CityWeather from './City/CityWeather';
import './styles/Weather.css';
import AfricaWeather from './AfricaWeather';

const Weather = () => {
  const [filteredCities, setFilteredCities] = useState([]);
  const cityData = useSelector((state) => state.weather.dataOfCities);
  const isDataFetched = useSelector((state) => state.weather.isDataFetched);
  const dispatch = useDispatch();

  const handleFilterChange = (filterText) => {
    const filtered = cityData.filter(
      (city) => city.city.toLowerCase().includes(
        filterText.toLowerCase(),
      ),
    );
    setFilteredCities(filtered);
  };

  useEffect(() => {
    if (!isDataFetched) {
      dispatch(fetchCitiesWeatherData());
    }
  }, [dispatch, isDataFetched]);

  useEffect(() => {
    setFilteredCities(cityData);
  }, [cityData]);

  return (
    <>
      <Nav onFilterChange={handleFilterChange} />
      <AfricaWeather />
      <div className="report-head">
        <p>City Reports</p>
      </div>
      <section className="main">
        {filteredCities.map((city) => (
          <Link className="city" to={`city/${city.id}`} key={city.id}>
            <CityWeather city={city.city} aqi={city.data.main.aqi} />
          </Link>
        ))}
      </section>

    </>
  );
};

export default Weather;

```